pulp_rpm/tests/functional/api/test_publish.py

```diff
@@ -72,6 +72,45 @@ def test_publish_any_repo_version(
             }
             rpm_publication_api.create(body)
 
+    @pytest.mark.parametrize("compression_type", ("gz", "zstd"))
+    @pytest.mark.parallel
+    def test_publish_with_compression_types(
+        self,
+        compression_type,
+        rpm_unsigned_repo_immediate,
+        rpm_publication_api,
+        gen_object_with_cleanup,
+        rpm_distribution_api,
+        monitor_task,
+    ):
+        """Sync and publish an RPM repository w/ zstd compression and verify it exists."""
+        # 1. Publish and distribute
+        publish_data = RpmRpmPublication(
+            repository=rpm_unsigned_repo_immediate.pulp_href, compression_type="zstd"
+        )
+        publish_response = rpm_publication_api.create(publish_data)
+        created_resources = monitor_task(publish_response.task).created_resources
+        publication_href = created_resources[0]
+
+        body = gen_distribution(publication=publication_href)
+        distribution = gen_object_with_cleanup(rpm_distribution_api, body)
+
+        # 2. check "primary", "filelists", "other" and "updatedinfo" are in .zstd format
+        repomd = ElementTree.fromstring(
+            requests.get(os.path.join(distribution.base_url, "repodata/repomd.xml")).text
+        )
+        xpath_data = "{{{}}}data".format(RPM_NAMESPACES["metadata/repo"])
+        xpath_location = "{{{}}}location".format(RPM_NAMESPACES["metadata/repo"])
```

Comment:
We can refactor this along with `_get_updateinfo_xml_path` into a shared helper function that returns the "data" value (primary, filelists, other, updateinfo, etc.) as dictionary keys and the href as dictionary values.

```python
def get_repomd_metadata_urls(self, repomd_url: str):
    repomd_xml = requests.get(os.path.join(distribution.base_url, "repodata/repomd.xml")).text
    repomd = ElementTree.fromstring(repomd_xml)    
    xpath_data = "{{{}}}data".format(RPM_NAMESPACES["metadata/repo"])
    xpath_location = "{{{}}}location".format(RPM_NAMESPACES["metadata/repo"])
    hrefs = {}
    for elem in repomd.findall(xpath_data):
        md_type = elem.get("type")
        hrefs[md_type] = elem.find(xpath_location).get("href")
    return hrefs
```

```python
# usage
for md_type, location in self.get_repomd_metadata_urls(url).items():
    if md_type in {"primary", "filelists", "other":
        assert location.endswith(".zst")
```

Reply:
Thanks for snippet